upgrade/upgrade-1.2.12.php

```diff
@@ -30,5 +32,42 @@
  */
 function upgrade_module_1_2_12($module)
 {
-    return $module->registerHook('displayInvoiceLegalFreeText');
+    foreach (PrestaShop\Module\PrestashopCheckout\OrderStates::ORDER_STATES as $state => $color) {
+        $orderStateId = (int) \Configuration::get($state);
+        setStateIcons($state, $orderStateId, $module->name);
```

Comment:
we can, @Matt75 and me are for it, what do you think ?

Reply:
@Matt75 @zalexki finaly, as the method `setStateIcons` might be updated in further monthes, we should use it for the upgrade. If the merchant update his module in 1 year we won't have the expected update. 
I think that "duplicating" the code for the upgrade is the best solution. It will never change and we we'll be certain that the result won't change.

Reply:
We're partially using the OrderStates class, for instance with its contants. We may go further and use its methods in charge of setting the proper color.

I'm resolving this conversation as we're discussing about it in another one.